@dstebila @xvzcf When compiling as a shared library, I'm getting a whole bunch of OpenSSL-related linker errors at the end (although OpenSSL is properly detected and enabled on my side), like

[ 98%] Linking C shared library ../lib/liboqs.dylib
Undefined symbols for architecture x86_64:
  "_BN_CTX_end", referenced from:
      _OQS_KEM_bike1_l1_cpa_ossl_add in openssl_utils.c.o
      _OQS_KEM_bike1_l1_cpa_cyclic_product in openssl_utils.c.o
      _OQS_KEM_bike1_l3_cpa_ossl_add in openssl_utils.c.o
      _OQS_KEM_bike1_l3_cpa_cyclic_product in openssl_utils.c.o
      _OQS_KEM_bike1_l1_fo_ossl_add in openssl_utils.c.o
      _OQS_KEM_bike1_l1_fo_cyclic_product in openssl_utils.c.o
      _OQS_KEM_bike1_l3_fo_ossl_add in openssl_utils.c.o
      ...
  "_BN_CTX_free", referenced from:
      _OQS_KEM_bike1_l1_cpa_ossl_add in openssl_utils.c.o
      _OQS_KEM_bike1_l1_cpa_cyclic_product in openssl_utils.c.o
      _OQS_KEM_bike1_l3_cpa_ossl_add in openssl_utils.c.o
      _OQS_KEM_bike1_l3_cpa_cyclic_product in openssl_utils.c.o
      _OQS_KEM_bike1_l1_fo_ossl_add in openssl_utils.c.o
      _OQS_KEM_bike1_l1_fo_cyclic_product in openssl_utils.c.o
      _OQS_KEM_bike1_l3_fo_ossl_add in openssl_utils.c.o
      ...
  "_BN_CTX_get", referenced from:
      _OQS_KEM_bike1_l1_cpa_ossl_add in openssl_utils.c.o
      _OQS_KEM_bike1_l1_cpa_cyclic_product in openssl_utils.c.o
      _OQS_KEM_bike1_l3_cpa_ossl_add in openssl_utils.c.o
      _OQS_KEM_bike1_l3_cpa_cyclic_product in openssl_utils.c.o
      _OQS_KEM_bike1_l1_fo_ossl_add in openssl_utils.c.o
      _OQS_KEM_bike1_l1_fo_cyclic_product in openssl_utils.c.o
      _OQS_KEM_bike1_l3_fo_ossl_add in openssl_utils.c.o
      ...
  "_BN_CTX_new", referenced from:
      _OQS_KEM_bike1_l1_cpa_ossl_add in openssl_utils.c.o
      _OQS_KEM_bike1_l1_cpa_cyclic_product in openssl_utils.c.o
      _OQS_KEM_bike1_l3_cpa_ossl_add in openssl_utils.c.o
      _OQS_KEM_bike1_l3_cpa_cyclic_product in openssl_utils.c.o
      _OQS_KEM_bike1_l1_fo_ossl_add in openssl_utils.c.o
      _OQS_KEM_bike1_l1_fo_cyclic_product in openssl_utils.c.o
      _OQS_KEM_bike1_l3_fo_ossl_add in openssl_utils.c.o

Any idea?

--- a/.travis.yml
+++ b/.travis.yml
@@ -16,7 +16,7 @@ matrix:
       before_install:
         - rm -rf /usr/local/include/c++
         - brew update
-        - brew install gcc6 cmake ninja
+        - brew install gcc6 ninja
 
     # OSX gcc6
     - os: osx
@@ -25,7 +25,7 @@ matrix:
       before_install:
         - rm -rf /usr/local/include/c++
         - brew update
-        - brew install gcc6 cmake ninja
+        - brew install gcc6 ninja
         - export CXX="g++-6" CC="gcc-6"
         - sudo softwareupdate -i "Command Line Tools (macOS High Sierra version 10.13) for Xcode-10.1"
 

--- a/.travis/install_liboqs.sh
+++ b/.travis/install_liboqs.sh
@@ -6,4 +6,4 @@ git checkout master
 mkdir build && cd build
 cmake -GNinja -DBUILD_SHARED_LIBS=ON ..
 ninja
-ninja install
+sudo ninja install

--- a/README.md
+++ b/README.md
@@ -58,7 +58,7 @@ directory to `build`, then type
 	make -j4
 
 The above commands build all examples in `examples`, i.e. `examples/kem` and `examples/sig`, assuming
-the CMake build system is available on your platform. The `-DLIBOQS_INCLUDE_DIR` and `-DLIBOQS_LIB_DIR` flags specify the location to the liboqs headers and compiled library. You may omit those flags and simply type `cmake .. && make -j4` in case you installed liboqs in `/usr/local` (true if you ran `sudo make install` after building liboqs). You may replace the `-j4` flag with your
+the CMake build system is available on your platform. The `-DLIBOQS_INCLUDE_DIR` and `-DLIBOQS_LIB_DIR` flags specify the location to the liboqs headers and compiled library. You may omit those flags and simply type `cmake .. && make -j4` in case you installed liboqs in `/usr/local` (true if you ran `sudo ninja install` after building liboqs). You may replace the `-j4` flag with your
 processor's number of cores, e.g. use `-j8` if your system has 8 cores.
 To build only a specific example, e.g. `examples/kem`, specify the target as the argument of the `make` command, such as
 

